test/utils.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright 2017 - Edoardo Morassutto <edoardo.morassutto@gmail.com>
+import os
+import string
+import tempfile
+import contextlib
+import sys
+
+import datetime
+
+import random
+
+from src.config import Config
+from src.database import Database
+from src.logger import Logger
+
+
+class Utils:
+    prefix = None
+
+    @staticmethod
+    def prepare_test(load_config=True, connect_database=True, connect_logger=True):
+        config_file_name = Utils.new_tmp_file()
+        log_file_name = Utils.new_tmp_file()
+        db_file_name = Utils.new_tmp_file()
+
+        with open(config_file_name, 'w') as file:
+            file.write("logfile: %s\n"
+                       "db: %s\n"
+                       "storedir: %s\n" % (log_file_name, db_file_name, Utils.new_tmp_dir()))
+
+        if load_config:
+            Config.loaded = False
+            Config.set_config_file(config_file_name)
+        if connect_logger:
+            Logger.connected = False
+            Logger.connect_to_database()
+            Logger.set_log_level("WARNING")
+        if connect_database:
+            Database.connected = False
+            Database.connect_to_database()
+
+    @staticmethod
+    @contextlib.contextmanager
+    def nostderr():
+        """
+        Suppress stderr output in the with context
+        http://stackoverflow.com/a/1810086
+        """
+        savestderr = sys.stderr
+        class Devnull(object):
+            def __init__(self): self.buffer = ""
+            def write(self, data): self.buffer += data
+            def flush(self): pass
+        sys.stderr = Devnull()
+        try:
+            yield sys.stderr
+        finally:
+            sys.stderr = savestderr
+
+    @staticmethod
+    def get_tmp_dir():
+        if Utils.prefix is None:
+            Utils.prefix = os.path.join(tempfile.gettempdir(), "territoriali-backend-tests",
+                                        datetime.datetime.now().strftime("temp-%Y-%m-%d_%H-%M-%S"))
```

Comment:
Each run of `./run_tests.py` has it's temp files in an isolated folder. You should be able to debug each run a bit better... Also you won't have a directory in you fs with thousands of files.

`gettempdir` should point to a valid temporary directory, the current time acts like an id for the run of the tests